docs/src/Dropdown.doc.js

```diff
@@ -543,59 +961,232 @@ function SubtextIconButtonPopoverExample() {
+    >
+      <MainSection.Card
+        cardSize="lg"
+        defaultCode={`
+function OnNavigation() {
+  const [clientOnNavigationMode, setClientOnNavigationMode] = React.useState('navigation');
+  const [open, setOpen] = React.useState(false);
+  const [selected, setSelected] = React.useState(null);
+  const anchorRef = React.useRef(null);
+  const handleSelect = ({item}) => {
+    setSelected(item);
+  };
 
-card(
-  <Example
-    id="withCustomItemContent"
-    name="With Custom Item Content"
-    description={`
+  const onNavigation = ({ href, onNavigationOptions }) => {
+
+    const onNavigationClick = ({ event }) => {
+      event.nativeEvent.preventDefault();
+      // eslint-disable-next-line no-alert
+      alert('Disabled link: '+href+'. Opening help.pinterest.com instead.');
+      window.open('https://help.pinterest.com', '_blank');
+    }
+
+    return onNavigationOptions && onNavigationOptions.navigationMode === 'navigation'
+      ? onNavigationClick
+      : null;
+  }
+
+  const linkProps = {
+    href: 'https://pinterest.com',
+    onNavigationOptions: {
+      navigationMode: clientOnNavigationMode
+    },
+    target: 'blank',
+  }
+
+  return (
+    <Provider onNavigation={onNavigation}>
+      <Flex direction="column" gap={2}>
+        <Flex direction="column" gap={2}>
+          <Text>Navigation type:</Text>
+          <RadioButton
+            checked={clientOnNavigationMode === 'link'}
+            id="link3"
+            label="Default Link"
+            name="navigation3"
+            onChange={() => setClientOnNavigationMode('link')}
+            value="link"
+          />
+          <RadioButton
+            checked={clientOnNavigationMode === 'navigation'}
+            id="navigation3"
+            label="Custom Navigation Context"
+            name="navigation"
+            onChange={() => setClientOnNavigationMode('navigation')}
+            value="navigation"
+          />
+          <Divider/>
+        </Flex>
+        <Box display="flex" justifyContent="center">
+          <Button
+            accessibilityControls="custom-nav-dropdown-example"
+            accessibilityHaspopup
+            accessibilityExpanded={open}
+            iconEnd="arrow-down"
+            text="Menu"
+            inline
+            ref={anchorRef}
+            selected={open}
+            onClick={ () => setOpen((prevVal) => !prevVal) }
+          />
+          {open && (
+            <Dropdown id="custom-nav-dropdown-example" anchor={anchorRef.current} onDismiss={() => {setOpen(false)}}>
+              <Dropdown.Item
+                { ...linkProps }
+                isExternal
+                option={{ value: 'item 3', label: 'Item 3 with a really long, detailed, complex name' }}
+              />
+            </Dropdown>
+          )}
+        </Box>
+      </Flex>
+    </Provider>
+  );
+}
+`}
+      />
+    </MainSection.Subsection>
+    <MainSection.Subsection
+      title="Custom item content"
```

Comment:
How do you feel about `This can be useful when extra functionality is needed beyond a basic Link, such as supporting React Router`? I think it might conflict with the new onClick prop that I added.